### ISSUE TITLE
VALUES is really just LIST

### ISSUE THREAD

**Issue by heegaiximephoomeeghahyaiseekh** [shikantaza/pLisp#33]
While being similar to lists, multiple return values in Lisp are supposed to be a separate type that allows you to implicitly ignore the additional values. For example, you should be able to do this:

    (car (values '(1 2) '(3 4))) => 1

In my (unpublished) interpreter, I handle this by implementing a special type, TUPLE, which is produced only by VALUES. The TUPLE type can be just a list or vector with a different type tag.

Whenever a TUPLE passes to anything other than `MULTIPLE-VALUE-LIST`, it should be converted to the first value in the tuple. Other multiple-value forms, such as `MULTIPLE-VALUE-BIND` and `MULTIPLE-VALUE-CALL`, can be implemented as macros.

**Comment by shikantaza** [comment]
I'm somewhat ambivalent about this, this would involve adding a new object type as you mentioned, and may need extensive changes. I'll investigate this further. Deviating from Lisp canon seems to be the easier option here :-)

**Comment by j3pic** [comment]
Maybe just removing VALUES would be the thing to do here. Why have something with that name if the concept is not actually supported?

**Comment by shikantaza** [comment]
Yeah, it's an artifact of an earlier thought process. I'm still trying to
remain as close as possible to Common Lisp though, will investigate what it
takes to implement VALUES faithfully; will remove it if not feasible given
pLisp's object model.

Regards,
Rajesh
